src/Reflection/Annotations/AnnotationsMethodsClassReflectionExtension.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\Annotations;
+
+use PHPStan\Broker\Broker;
+use PHPStan\Reflection\BrokerAwareClassReflectionExtension;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\MethodsClassReflectionExtension;
+use PHPStan\Type\FileTypeMapper;
+use PHPStan\Type\VoidType;
+
+class AnnotationsMethodsClassReflectionExtension implements MethodsClassReflectionExtension, BrokerAwareClassReflectionExtension
+{
+
+	/** @var FileTypeMapper */
+	private $fileTypeMapper;
+
+	/** @var Broker */
+	private $broker;
+
+	/** @var MethodReflection[][] */
+	private $methods = [];
+
+	public function __construct(FileTypeMapper $fileTypeMapper)
+	{
+		$this->fileTypeMapper = $fileTypeMapper;
+	}
+
+	public function setBroker(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	public function hasMethod(ClassReflection $classReflection, string $methodName): bool
+	{
+		if (!isset($this->methods[$classReflection->getName()])) {
+			$this->methods[$classReflection->getName()] = $this->createMethods($classReflection);
+		}
+
+		return isset($this->methods[$classReflection->getName()][$methodName]);
+	}
+
+	public function getMethod(ClassReflection $classReflection, string $methodName): MethodReflection
+	{
+		return $this->methods[$classReflection->getName()][$methodName];
+	}
+
+	/**
+	 * @param ClassReflection $classReflection
+	 * @return MethodReflection[]
+	 */
+	private function createMethods(ClassReflection $classReflection): array
+	{
+		$methods = [];
+		foreach ($classReflection->getParentClassesNames() as $parentClassName) {
+			if (!$this->broker->hasClass($parentClassName)) {
+				continue;
+			}
+			$methods += $this->createMethods($this->broker->getClass($parentClassName));
+		}
+
+		$fileName = $classReflection->getNativeReflection()->getFileName();
+		if ($fileName === false) {
+			return $methods;
+		}
+
+		$docComment = $classReflection->getNativeReflection()->getDocComment();
+		if ($docComment === false) {
+			return $methods;
+		}
+
+		$typeMap = $this->fileTypeMapper->getTypeMap($fileName);
+
+		preg_match_all('#@method\s+(?:(.*)\s+)?([a-zA-Z0-9_]+)\(.*\)#', $docComment, $matches, PREG_SET_ORDER);
+		foreach ($matches as $match) {
+			$typeStringCandidate = $match[1];
+			if (preg_match('#' . FileTypeMapper::TYPE_PATTERN . '#', $typeStringCandidate, $typeStringMatches)) {
+				$typeString = $typeStringMatches[1];
+				if (!isset($typeMap[$typeString])) {
+					continue;
+				}
+				$returnType = $typeMap[$typeString];
+			} else {
+				$returnType = new VoidType();
```

Comment:
Ok. It could be MixedType. But according to [phpDocumentor](https://phpdoc.org/docs/latest/references/phpdoc/tags/method.html): "When the intended method does not have a return value then the return type MAY be omitted; in which case ‘void’ is implied."
What do you think? The final decision is up to you...